native/libcst/src/tokenizer/whitespace_parser.rs

```diff
@@ -74,12 +74,44 @@ impl<'a> Config<'a> {
                 break;
             }
         }
-        let default_newline =
-            NEWLINE_RE_2.with(|r| r.find(input).map(|m| m.as_str()).unwrap_or("\n"));
+
+        let mut lines = Vec::new();
```

Comment:
A possible optimisation is to initialize the Vec with `input.len() / 88` (or 100 or 120, depending on how defensive we want to be). This turned out to improve performance for ruff.